src/sentry/models/event.py

```diff
@@ -264,8 +266,46 @@ def checksum(self):
         return ''
 
     def get_email_subject(self):
-        return '[%s] %s: %s' % (
-            self.project.get_full_name().encode('utf-8'),
-            six.text_type(self.get_tag('level')).upper().encode('utf-8'),
-            self.title.encode('utf-8')
-        )
+        template = self.project.get_option('mail:subject_template')
+        if template:
+            template = EventTemplate(template)
+        else:
+            template = DEFAULT_SUBJECT_TEMPLATE
+        return truncatechars(template.render(self).encode('utf-8'), 128)
+
+
+class EventTemplate(string.Template):
```

Comment:
Subclassing `string.Template` here seems a little hacky to me.

In my opinion, it'd be simpler to replace this with an object that acted as the data mapping expected by the existing `Template.safe_substitute` API that's being more or less emulated here:

```python
class EventTemplateData(object):
    def __init__(self, event):
        self.event = event

    def __getitem__(self, name):
        if name == 'project':
            return self.event.project.get_full_name()
        elif name == 'projectID':
            return self.event.project.slug
        elif name == 'orgID':
            return self.event.organization.slug
        elif name == 'level':
            return six.text_type(self.event.get_tag('level')).upper()
        elif name == 'environment':
            return six.text_type(self.event.get_tag('environment') or '')
        elif name == 'release':
            return six.text_type(self.event.get_tag('sentry:release') or '')
        elif name == 'title':
            return self.event.title
        elif name == 'transaction':
            return six.text_type(self.event.get_tag('transaction') or '')
        else:
            raise KeyError

template = string.Template('[$project] $level: $title')
data = EventTemplateData(event)
subject = template.safe_substitute(data)
```

Also, would it make sense to just allow arbitrary tag names here, since people seem to ask for that relatively often?